### ISSUE TITLE
Axios includes br in Accept-Encoding even if transparent decompression is not supported

### ISSUE THREAD

**Issue by lucaswerkmeister** [axios/axios#5258]
### Describe the bug

Axios has optional support for transparently decompressing brotli-encoded responses, depending on whether brotli support is available in the runtime.

```js
const isBrotliSupported = utils.isFunction(zlib.createBrotliDecompress);
```

```js
        case 'br':
          if (isBrotliSupported) {
            streams.push(zlib.createBrotliDecompress());
            delete res.headers['content-encoding'];
          }
```

However, it unconditionally includes `br` in the `Accept-Encoding` header it sets (if the header wasn’t specified by the library user), regardless of whether the runtime supports brotli or not.

```js
    headers.set('Accept-Encoding', 'gzip, deflate, br', false);
```

So if brotli support isn’t available, and the server decides to encode the response using brotli (which it’s allowed to, based on the header axios added), then the library user will be left with a response they can’t decode, even though they don’t asked for brotli encoding.

### Axios Version

1.2.0-alpha.1

### Adapter Version

http

**Comment by mcuelenaere** [comment]
Isn't this fixed since https://github.com/axios/axios/pull/5306?

**Comment by lucaswerkmeister** [comment]
True, I forgot to close this when I closed #5259.
